lib/modules/@apostrophecms/utils/src/apos/always/http.js

```diff
@@ -120,36 +124,21 @@
         url += encodeURIComponent(key) + '=' + encodeURIComponent(data[key]);
       }
     }
-    xmlhttp.open(method, url);
-    var formData = window.FormData && (data instanceof window.FormData);
-    var sendJson = (options.send === 'json') || (options.body && ((typeof options.body) === 'object') && !formData);
-    if (sendJson) {
-      xmlhttp.setRequestHeader('Content-Type', 'application/json');
-    }
-    if (csrfToken && (options.csrf !== false)) {
-      if (apos.utils.sameSite(url)) {
-        xmlhttp.setRequestHeader('X-XSRF-TOKEN', csrfToken);
-      }
-    }
-    if (options.headers) {
-      var keys = Object.keys(options.headers);
-      for (i = 0; (i < keys.length); i++) {
-        xmlhttp.setRequestHeader(keys[i], options.headers[keys[i]]);
-      }
-    }
+
     apos.utils.emit(document.body, 'apos-before-' + method.toLowerCase(), {
       uri: url,
       data: options.body,
       request: xmlhttp
     });
-    if (sendJson) {
-      data = JSON.stringify(options.body);
-    } else {
-      data = options.body;
-    }
-    xmlhttp.send(data);
-    xmlhttp.addEventListener("load", function() {
-      var responseHeader = this.getResponseHeader("Content-Type");
+
+    xmlhttp.addEventListener('progress', function() {
```

Comment:
I moved all the listeners here (before the `open` and `send` methods) in order to have the `progress` listener work correctly (https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Using_XMLHttpRequest?redirectlocale=en-US&redirectslug=DOM/XMLHttpRequest/Using_XMLHttpRequest#Monitoring_progress)